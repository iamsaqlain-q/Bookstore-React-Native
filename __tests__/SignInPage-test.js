import React from 'react';
import renderer from 'react-test-renderer';
import SignInPage from '../src/utility/SignInPage';
import {render, fireEvent} from '@testing-library/react-native';

test('SignIn renders correctly', () => {
  const element = renderer.create(<SignInPage />).toJSON();
  expect(element).toMatchSnapshot();
});

it('renders SignIn', () => {
  const element = renderer.create(<SignInPage />).toJSON();
  expect(element).toMatchSnapshot();
});

test('TextInput test', () => {
  const {getByTestId} = render(<SignInPage />);
  const input = getByTestId('emailInput');
  fireEvent.changeText(input, 'abc@gmail.com');
  expect(input.props.value).toBe('abc@gmail.com');
});
