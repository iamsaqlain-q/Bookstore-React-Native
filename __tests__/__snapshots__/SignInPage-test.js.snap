// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FFC0CB",
        "borderRadius": 30,
        "height": 45,
        "marginBottom": 20,
        "width": "70%",
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Email."
      placeholderTextColor="#003f5c"
      style={
        {
          "flex": 1,
          "height": 50,
          "marginLeft": 20,
          "padding": 10,
        }
      }
      testID="emailInput"
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FFC0CB",
        "borderRadius": 30,
        "height": 45,
        "marginBottom": 20,
        "width": "70%",
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Password."
      placeholderTextColor="#003f5c"
      secureTextEntry={true}
      style={
        {
          "flex": 1,
          "height": 50,
          "marginLeft": 20,
          "padding": 10,
        }
      }
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        {
          "height": 30,
          "marginBottom": 30,
        }
      }
    >
      Forgot Password?
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FF1493",
        "borderRadius": 25,
        "height": 50,
        "justifyContent": "center",
        "marginTop": 40,
        "opacity": 1,
        "width": "80%",
      }
    }
  >
    <Text>
      SignIn
    </Text>
  </View>
</View>
`;

exports[`renders SignIn 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FFC0CB",
        "borderRadius": 30,
        "height": 45,
        "marginBottom": 20,
        "width": "70%",
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Email."
      placeholderTextColor="#003f5c"
      style={
        {
          "flex": 1,
          "height": 50,
          "marginLeft": 20,
          "padding": 10,
        }
      }
      testID="emailInput"
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FFC0CB",
        "borderRadius": 30,
        "height": 45,
        "marginBottom": 20,
        "width": "70%",
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Password."
      placeholderTextColor="#003f5c"
      secureTextEntry={true}
      style={
        {
          "flex": 1,
          "height": 50,
          "marginLeft": 20,
          "padding": 10,
        }
      }
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        {
          "height": 30,
          "marginBottom": 30,
        }
      }
    >
      Forgot Password?
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FF1493",
        "borderRadius": 25,
        "height": 50,
        "justifyContent": "center",
        "marginTop": 40,
        "opacity": 1,
        "width": "80%",
      }
    }
  >
    <Text>
      SignIn
    </Text>
  </View>
</View>
`;
